Libraries/Dijkstra/Dijkstra.cpp

```diff
@@ -0,0 +1,166 @@
+        matrix_[current_coord_->x + 1][current_coord_->y].prev = matrix_[current_coord_->x][current_coord_->y].current;
+      }
+    }
+    if (tiles_map.getTile(current_coord_->x, current_coord_->y).ableToGoWest()
+    && !tiles_map.getTile(current_coord_->x, current_coord_->y - 1).isBlack()) {
+      if (tiles_map.getTile(current_coord_->x, current_coord_->y - 1).isVisited()
+      && !matrix_[current_coord_->x][current_coord_->y - 1].checked) {
+        open_vector_.pushAsLast(&matrix_[current_coord_->x][current_coord_->y - 1].current);
+        matrix_[current_coord_->x][current_coord_->y - 1].checked = true;
+      }
+      if (matrix_[current_coord_->x][current_coord_->y - 1].weight > matrix_[current_coord_->x][current_coord_->y].weight + tiles_map.getTile(current_coord_->x, current_coord_->y - 1).getWeight()) {
+        matrix_[current_coord_->x][current_coord_->y - 1].weight = matrix_[current_coord_->x][current_coord_->y].weight + tiles_map.getTile(current_coord_->x, current_coord_->y - 1).getWeight();
+        matrix_[current_coord_->x][current_coord_->y - 1].prev = matrix_[current_coord_->x][current_coord_->y].current;
+      }
+    }
+    open_vector_.popFirst();
+  }
+}
+
+Dijkstra::Dijkstra(const Map tiles_map) {
+  initialValues(tiles_map);
+
+  for (uint8_t row = 0; row < tiles_map.numberOfRows(); ++row) {
+    for (uint8_t column = 0; column < tiles_map.numberOfColumns(); ++column) {
+      if (tiles_map.getTile(row, column).isAccessible() &&
+      !tiles_map.getTile(row, column).isVisited() &&
+      !tiles_map.getTile(row, column).isBlack()) {
+        current_coord_ = &matrix_[row][column].current;
+        non_visited_vector_.pushAsLast(current_coord_);
+      }
+    }
+  }
+}
+
+Dijkstra::Dijkstra(const Map tiles_map, Tile tile_to_find) {
+  initialValues(tiles_map);
+}
+
+void Dijkstra::printDijkstraMatrix(const Map tiles_map) {
+  for (int row = 0 ; row < tiles_map.numberOfRows() ; ++row) {
+    for (int column = 0 ; column < tiles_map.numberOfColumns() ; ++column) {
+      Serial.print("(");
+      Serial.print(matrix_[row][column].current.x);
+      Serial.print(",");
+      Serial.print(matrix_[row][column].current.y);
+      Serial.print(")");
+      Serial.print(matrix_[row][column].weight);
+      Serial.print("  ");
+    }
+    Serial.println("");
+  }
+}
+
+TVector<char> Dijkstra::getPath(const Map tiles_map) {
+  updateMatrix(tiles_map);
+  if (non_visited_vector_.getSize() > 0) current_coord_ = non_visited_vector_[0];
+  else Serial.print("There are no Tiles to travel");
+
+  // Comparing all the non-visited tiles.
+  for (int i = 1 ; i < non_visited_vector_.getSize() ; ++i) {
+    if (matrix_[current_coord_->x][current_coord_->y].weight > matrix_[non_visited_vector_[i]->x][non_visited_vector_[i]->y].weight) {
+      current_coord_ = non_visited_vector_[i];
+    }
+  }
+
+  printDijkstraMatrix(tiles_map);
+
+  while (current_coord_->x != current_row_ || current_coord_->y != current_column_) {
```

Comment:
Yes, because the objective is that 
current_coord_->x == current_row_ AND current_coord_->y == current_column_,
so the process is going to repeat until that happens.